src/stores/utilities/pacbioLibraries.js

```diff
@@ -30,28 +30,23 @@ const validateLibraryFields = (library) => {
  * Formats and transforms libraries.
  *
  * @param {Object} libraries - The libraries to format and transform.
- * @param {Object} tubes - The tubes associated with the libraries.
  * @param {Object} tags - The tags associated with the libraries.
  * @param {Object} requests - The requests associated with the libraries.
  * @returns {Array<Object>} - The formatted and transformed libraries.
  */
-const formatAndTransformLibraries = (libraries, tubes, tags, requests) =>
-  Object.values(libraries)
-    .filter((library) => library.tube)
-    .map((library) => {
-      const { id, request, tag_id, tag, tube, ...attributes } = library
-      const tagId = tag_id ?? tag
-      const tagGroupId = tags[tagId] ? (tags[tagId].group_id ?? '') : ''
-      return {
-        id,
-        tag_id: String(tagId),
-        tube,
-        ...attributes,
-        tag_group_id: tagGroupId,
-        sample_name: requests[request]?.sample_name,
-        barcode: tubes[tube]?.barcode,
-      }
-    })
+const formatAndTransformLibraries = (libraries, tags, requests) =>
+  Object.values(libraries).map((library) => {
+    const { id, request, tag_id, tag, ...attributes } = library
+    const tagId = tag_id ?? tag
+    const tagGroupId = tags[tagId] ? (tags[tagId].group_id ?? '') : ''
```

Comment:
The above two lines also feels like a code smell with inconsistent data structure.

Reply:
Yes I think it is accounting for various library structures. Ones with tags, one with tag_ids and ones without tags. Ill add a TODO comment to look into it.